packages/python/plotly/plotly/basedatatypes.py

```diff
@@ -1302,7 +1302,7 @@ def _add_annotation_like(
             # if exclude_empty_subplots is True, check to see if subplot is
             # empty and return if it is
             if exclude_empty_subplots and (
-                not self._subplot_contains_trace(xref, yref)
+                not self._subplot_not_empty(xref, yref, selector=exclude_empty_subplots)
```

Comment:
does this line mean that one could do `fig.add_shape( ... , row="all", col="all", exclude_empty_subplots=["traces", "shapes"])` ?

Reply:
to be clear: i don't think that's necessary, I would just do `True`/`False` here :)

Reply:
You could but by default True means "all". I also don't think it's necessary but you get this functionality for free, all that needs to happen is the list of objects to check is made into the keyword argument of the function (and coerced to "all" if it is True). But I can disable that if you want and have it just accept True/False.

Reply:
Yeah, let's just accept truthy/falsey for now. I think it's easier to not support certain exotic use-cases that then some people depend upon later :)